### PR TITLE
Use rfcbot style for FCPs in the prioritization agenda

### PR DESCRIPTION
**Pull request by Urgau**
This PR add the necessary information to replicate the rfcbot style when showing FCPs and uses it in the prioritization agenda.

~~One thing at isn't perfect is that we are using the Github login and not the Zulip login, using the Zulip username would require accessing the team database which I didn't wanted to do yet but could be done as a follow up PR.~~ (fixed, https://github.com/rust-lang/triagebot/pull/1784#issuecomment-2016833448)

r? @apiraino 
 
--------

They would now look like this (modulo the mentions). ~~Full example [here](https://hackmd.io/@Urgau/rytauC3Aa).~~

- merge: [Add a new `--build-id` flag to rustc (compiler-team#635)](https://github.com/rust-lang/compiler-team/issues/635#issuecomment-1777545767)
    * `@wesleywiser` `@davidtwco` `@Aaron1011` `@petrochenkov`
    * [other-existing-options (by petrochenkov)](https://github.com/rust-lang/compiler-team/issues/635#issuecomment-1779597156) [option-name (by wesleywiser)](https://github.com/rust-lang/compiler-team/issues/635#issuecomment-1785349936)
- merge: [Retire the mailing list and make all decisions on zulip (compiler-team#649)](https://github.com/rust-lang/compiler-team/issues/649#issuecomment-1618781780)
    * no pending checkboxs
    * [automatic-sync (by compiler-errors)](https://github.com/rust-lang/compiler-team/issues/649#issuecomment-1618902660) [single-point-of-failure-via-stream-archival (by pnkfelix)](https://github.com/rust-lang/compiler-team/issues/649#issuecomment-1664056989)

### CODE REVIEW

**Comment by Urgau** [comment]
I've found a pretty neat trick. Zulip as a de-ambiguator syntax when pinging some, it works by specifying the Zulip ID after the name, like so `@**Urgau|123456**`; but if we omit the name and just provide the ID it will ping the right person. Simplifying the greatly the process.

I've added this is the last commit.

File: src/actions.rs [hunk 1]
```diff
@@ -140,7 +156,7 @@ impl<'a> Action for Step<'a> {
                         let issues = query
                             .query(
                                 &repository,
-                                name == "proposed_fcp",
+                                name.contains("fcp"),
```
Comment (apiraino) [inline src/actions.rs:159]: I would probably revert this change. It will run an additional HTTP query also for FCPs that don't need the additional details. In fact we're interested to show these details only for FCPs with label `proposed-final-comment-period`.

File: src/github.rs [hunk 2]
```diff
@@ -1750,6 +1758,42 @@ impl<'q> IssuesQuery for Query<'q> {
                         bot_tracking_comment_content,
                         initiating_comment_html_url: init_comment.html_url.clone(),
                         initiating_comment_content: quote_reply(&init_comment.body),
+                        disposition: fcp
+                            .fcp
+                            .disposition
+                            .as_deref()
+                            .unwrap_or("<unknown>")
+                            .to_string(),
+                        pending_reviewers: fcp
```
Comment (apiraino) [inline src/github.rs:1767]: TIL using the Zulip ID for pings, that's a neat trick! 

Though I think in this context we don't want to emit ping every week, I'm afraid it will just become annoying. I would rather instead remove the `@` mentions and just list the people the FCP is waiting on.

File: templates/prioritization_agenda.tt
Comment (apiraino) [inline templates/prioritization_agenda.tt:1]: I think the new Jinja template needs some tweaks to look like the current Markdown we produce (a bit more compact).

Also, the changes to the headings in this template `prioritization_agenda.tt` could probably be reverted (again, to keep the template as compact as possible)

**Review by apiraino — CHANGES_REQUESTED** [review]
hey @Urgau thanks for working on this! :slightly_smiling_face: 

I like the new layout you suggest for the FCPs.

I left a few comments (mostly nits), let me know what you think!

File: templates/prioritization_agenda.tt
Comment (Urgau) [reply templates/prioritization_agenda.tt:1]: IMO the current template is already too compact and difficult to read. With this new style it became unreadable. I can try to make it more compact but it's already very compact.

File: src/github.rs (hunk 2 above)
Comment (Urgau) [reply src/github.rs:1767]: We may not want to ping every week but I think we should ping from time to time, and I think pinging T-compiler member in the meeting at the specific things waiting for them is acceptable to me.

Especially considering that they are already being pinging at the start of the meeting.

File: templates/prioritization_agenda.tt
Comment (Urgau) [reply templates/prioritization_agenda.tt:1]: After some discussions on Zulip, I decided to revert those stylistic changes.

Comment (Urgau) [reply templates/prioritization_agenda.tt:1]: One thing I just saw is that those bullet points are now so long that Zulip is collapsing the message by default. This is unfortunate.

File: src/github.rs (hunk 2 above)
Comment (Urgau) [reply src/github.rs:1781]: After some careful thinking I think I may have found a compromise.

I propose that we only mention under those circumstance:
  - we are 2 weeks into the FCP *(even if the are concerns registered, so that team members can voice there opinions on the FCP and concerns)*
  - or when there are no concerns and we are at least 4 weeks into the FCP *(no concerns + long after the start: probably didn't see the ping or forgot; and if the are not happy about the proposal they can just put up a concern)*

Btw, I used your `_` trick to do the no-but-still-looking-mention.

**Comment by apiraino** [comment]
Enabling mentions to remind team members checking their boxes in FCPs could have been a follow-up PR focused because it is unclear how it should work. However, for the sake of merging this change, I will approve this PR. For now it is disabled and will be discussed with T-compiler if and how that could work.

Thanks @Urgau for suggesting this formatting for the FCPs.

r=me

File: templates/prioritization_agenda.tt [hunk 3]
```diff
@@ -28,9 +29,9 @@ note_id: xxx
 - Old MCPs (not seconded, take a look)
 {{-issues::render(issues=mcp_old_not_seconded, indent="  ", with_age=true, empty="No old proposals this time.")}}
 - Pending FCP requests (check your boxes!)
-{{-issues::render(issues=in_pre_fcp, indent="  ", empty="No pending FCP requests this time.")}}
+{{-issues_rfcbot::render(issues=in_pre_fcp, indent="  ", empty="No pending FCP requests this time.")}}
 - Things in FCP (make sure you're good with it)
```
Comment (apiraino) [inline templates/prioritization_agenda.tt:33]: @Urgau The items in this section lost the links to the links to the pull requests

probably worth checking if your new macro is not eating other links around

**Review by apiraino — CHANGES_REQUESTED** [review]

File: templates/prioritization_agenda.tt (hunk 3 above)
Comment (Urgau) [reply templates/prioritization_agenda.tt:33]: Sorry for that, based on the extra information you gave in DM, I was able to fix it.

To make in short, when we don't have any details for an FCP we use a fallback and that fallback was manual instead of using the `_issue.tt` template 🤦

I pushed a commit using the proper fallback *as well as fixing a weird newline issue*.

File: src/github.rs [hunk 4]
```diff
@@ -1745,11 +1755,60 @@ impl<'q> IssuesQuery for Query<'q> {
...
+                    //  - or when there are no concerns and we are at least 4 weeks into the FCP.
+                    //
+                    // FIXME: This should get T-compiler approval before being enabled by default
+                    let should_mention = std::env::var("TRIAGEBOT_COMPILER_MENTION").is_ok() && {
+                        let now = chrono::offset::Utc::now();
+                        let time_diff = now - init_comment.created_at;
+                        time_diff.num_weeks() == 2
+                            || (time_diff.num_weeks() >= 4 && fcp.concerns.is_empty())
+                    };
+
```
Comment (apiraino) [inline src/github.rs:1769]:
```suggestion
                    // TODO: agree with the team(s) a policy to emit actual mentions to remind FCP
                    // voting member to cast their vote
                    let should_mention = false;
```

Comment (apiraino) [reply src/github.rs:1769]: @Urgau can we disable this for the moment and keep it as a reminder (which is useful)

Comment (Urgau) [reply src/github.rs]: I applied your patch *(which removes, not disable, it was already disabled)*.

I hope we can get a policy on this in the near future as IMO not having mention doesn't give much benefits.

Comment (apiraino) [reply src/github.rs]: yep, that's on my radar, don't worry :slightly_smiling_face: as I mentioned I just think that that needs a brief discussion involving people on the other end of these mentions and I don't want your work here to be blocked on that

**Comment by apiraino** [comment]
r=me

**Review by apiraino — APPROVED** [review]

**Comment by apiraino** [comment]
@jackh726 we're done here, I think. When you have a sec feel free to have a look (and ofc leave a comment). thanks!

File: templates/_issues_rfcbot.tt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{% import "_issue.tt" as issue %}
+
+{% macro render(issues, indent="", empty="None.") %}
+{%- for issue in issues %}
+{%- if issue.fcp_details is object %}
+{{indent}}- {{issue.fcp_details.disposition}}: [{{issue.title}} ({{issue.repo_name}}#{{issue.number}})]({{issue.fcp_details.bot_tracking_comment_html_url}})
+{{indent}}{{indent}}-{% for reviewer in issue.fcp_details.pending_reviewers %} @{% if issue.fcp_details.should_mention %}{% else %}_{% endif %}**|{{reviewer.zulip_id}}**{%else%} no pending checkboxs{% endfor %}
+{{indent}}{{indent}}-{% for concern in issue.fcp_details.concerns %} [{{concern.name}} (by {{concern.reviewer_login}})]({{concern.concern_url}}){%else%} no pending concerns{% endfor -%}
```
Comment (jackh726) [inline templates/_issues_rfcbot.tt:8]: It might be nice to write something like `concerns: ...`

**Review by jackh726 — COMMENTED** [review]
One small nit

**Review by jackh726 — APPROVED** [review]

File: templates/_issues_rfcbot.tt (hunk 5 above)
Comment (Urgau) [reply templates/_issues_rfcbot.tt]: Sure, pushed the change.
